actions/postgresql_show_create_table_action.go

```diff
@@ -0,0 +1,448 @@
+// pmm-agent
+// Copyright (C) 2018 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package actions
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"database/sql"
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/AlekSi/pointer"
+	_ "github.com/lib/pq" // register SQL driver
+	"github.com/percona/pmm/api/agentpb"
+	"github.com/pkg/errors"
+)
+
+type columnInfo struct {
+	Attname        string
+	FormatType     string
+	Substring      *string
+	Attnotnull     bool
+	Attcollation   *string
+	Attidentity    string
+	Attstorage     string
+	Attstattarget  *string
+	ColDescription *string
+}
+
+type indexInfo struct {
+	Relname            string
+	IsPrimary          bool
+	IsUnique           bool
+	IsClustered        bool
+	IsValid            bool
+	Indrelid           string
+	PgGetIndexDef      *string
+	PgGetConstraintDef *string
+	Contype            *string
+	Condeferrable      *bool
+	Condeferred        *bool
+	Indisreplident     *bool
+	Reltablespace      int
+}
+
+type postgresqlShowCreateTableAction struct {
+	id     string
+	params *agentpb.StartActionRequest_PostgreSQLShowCreateTableParams
+}
+
+// NewPostgreSQLShowCreateTableAction creates PostgreSQL SHOW CREATE TABLE Action.
+// This is an Action that can run `SHOW CREATE TABLE` command on PostgreSQL service with given DSN.
+func NewPostgreSQLShowCreateTableAction(id string, params *agentpb.StartActionRequest_PostgreSQLShowCreateTableParams) Action {
+	return &postgresqlShowCreateTableAction{
+		id:     id,
+		params: params,
+	}
+}
+
+// ID returns an Action ID.
+func (e *postgresqlShowCreateTableAction) ID() string {
+	return e.id
+}
+
+// Type returns an Action type.
+func (e *postgresqlShowCreateTableAction) Type() string {
+	return "postgresql-show-create-table"
+}
+
+// Run runs an Action and returns output and error.
+func (e *postgresqlShowCreateTableAction) Run(ctx context.Context) ([]byte, error) {
+	db, err := sql.Open("postgres", e.params.Dsn)
```

Comment:
Please use https://godoc.org/github.com/lib/pq#Connector and `sql.OpenDB` to use context cancelation for connections.